### ISSUE TITLE
Inconsistent handling of non-executable games

### ISSUE THREAD

**Issue by tobiasvl** [itchio/itch#2075]
I have made a game that's not an executable, but a Game Boy ROM file. It's not clear how to classify downloads for games like this on itch; it's a game, not a tool; it's not an executable, but it's playable on all platforms (as long as there is a Game Boy emulator for that platform). However, the classification is a minor "problem".

It's more problematic how these different downloads are handled by the app. I think there are several issues here, and this issue should perhaps be split.

* If I upload the ROM file and classify the download as "Other":
  * The itch app will say that it's not available for my platform, and not display a button to "Install" at all. **Expected behavior** would perhaps be to allow installation and showing a button to "Open" the folder, like it does for tools.

* If I upload the ROM file and classify the download as "Executable" (even though it's not an executable):
  * If I don't check any platform checkboxes, the itch app will say it's not available for my platform, and not display a button to "Install". (This is **expected behavior**, and is OK.)
  * If I check the checkbox for my platform (eg. "Windows"), the itch app will display an "Install" button. Clicking it does not work; it fails silently. **Expected behavior** would be to not fail, and act the same as it would for a `butler` build (see below).

* If I push the ROM file through `butler`:
  * If I check the checkbox for my platform (eg. "Windows"), the itch app will display an "Install" button. Clicking it installs the game successfully, and the button turns into a "Launch" button. Clicking this button gives an error, saying itch could not find an executable. **Expected behavior** would be to open the game's folder, like the case is for tools.

So, to sum up:

* The fact that the itch app treats a downloadable file differently from a `butler` build of the same file seems to be a bug.
* The fact that installing the regular downloadable file fails silently seems to be a bug.
* The fact that downloadable files classified as "Other" can't be installed and opened in the itch app is a feature request.

See also #1827.

**Comment by tobiasvl** [comment]
([This is the game](https://tobiasvl.itch.io/patricks-pocket-challenge), with both the directly uploaded ROM file and the `butler` build (to the channel `other`), in case you need to look at those to test the behavior.)

**Comment by fasterthanlime** [comment]
Thanks for the thorough report!

Here's itch v25's behavior with the page you just posted (page type is game, upload type is 'executable'): it lets you install it, then shows a Launch button, then it fails and shows this:

![image](https://user-images.githubusercontent.com/7998310/44929355-0f974080-ad5b-11e8-9b24-efa74f0b80b5.png)

I think this is the right behavior for Executable.

---

The problem I have is backend-related (/cc @leafo): when a page is of type 'Game', but only has uploads of type 'Other', it has no platforms at all. 

So:

  * The app's workaround to allow install *regardless of platforms* if it's neither a Game page or a Tool page doesn't kick in
  * It can't tell from the game API response alone that the game actually has an upload of type other that could very well be installed on macOS/Linux/Windows, and so, it should definitely show an install button.

I think one fix would be to treat "any" as a platform, and to include it in the API response for all games that have any non-executable upload available. Does that make sense @leafo @GranPC ?

---

Regarding this: 

> If I check the checkbox for my platform (eg. "Windows"), the itch app will display an "Install" button. Clicking it does not work; it fails silently. Expected behavior would be to not fail, and act the same as it would for a butler build (see below).

I think itch v23 (the current stable) would indeed silently fail to install a .gb file that isn't in a zip. As for v25, it currently fails too:

![image](https://user-images.githubusercontent.com/7998310/44929637-03f84980-ad5c-11e8-8d6a-0e03aeccd4f8.png)
![image](https://user-images.githubusercontent.com/7998310/44929649-0d81b180-ad5c-11e8-954f-b4bbfb4b5b37.png)

..but that's rather easy to fix. I just opened https://github.com/itchio/itch/issues/2076 to follow that.

**Comment by tobiasvl** [comment]
> Here's itch v25's behavior with the page you just posted (page type is game, upload type is 'executable'): it lets you install it, then shows a Launch button, then it fails and shows this
> I think this is the right behavior for Executable.

Yes, that behavior is indeed much better than what I'm seeing in v23. This is what I see there (the interface is in Norwegian, but it says "An error occured when starting Patrick's Pocket Challenge: No executable files found", and then no button to open the folder in explorer):

![image](https://user-images.githubusercontent.com/80824/44932594-72420980-ad66-11e8-8940-41d707e8f2e0.png)

> I think itch v23 (the current stable) would indeed silently fail to install a .gb file that isn't in a zip. As for v25, it currently fails too:

Aah, I mistakenly thought it was the fact that it wasn't a `butler` build that made it fail, but it's just the fact that it's not in a zip file. Makes more sense. And just to confirm that it does silently fail, here's the behavior I'm seeing:

![itch_app](https://user-images.githubusercontent.com/80824/44933288-d1088280-ad68-11e8-9252-30f1c89f0a68.gif)

(Just in case you want to test with my file, I've hid the `butler` build and unhidden the ROM file I originally uploaded in the web interface.)

Thanks for investigating!

**Comment by tobiasvl** [comment]
By the way, I don't suppose it would be possible to actually open the file?

It's just a Windows thing, I think, but if you double-click a non-executable in Windows, it will either open the program you have associated with that file type (which in my case would directly launch the Game Boy emulator I have installed and load the ROM), or open the "Open in..." dialog. I understand if that behavior could be confusing and/or insecure, but it'd be nice in the cases where the download is a single file.

**Comment by fasterthanlime** [comment]
> By the way, I don't suppose it would be possible to actually open the file?

That's a good question - I'm more worried about what would happen if you *don't* have any program to handle the file. But on Windows, we could probably query the registry to find that out - I'd probably put that behind a dialog, for the first time (with an "Always open with" checkbox), but that's a post-v25 feature :)

**Comment by fasterthanlime** [comment]
v25 is out and it has all the suggestions in this thread except "actually opening the file" - for which I recommend opening a separate issue.
